controllers/metallb_controller_test.go

```diff
@@ -59,6 +59,20 @@ var _ = Describe("MetalLB Controller", func() {
 			Expect(speakerDaemonSet).NotTo(BeZero())
 			Expect(len(speakerDaemonSet.Spec.Template.Spec.Containers)).To(BeNumerically(">", 0))
 			Expect(speakerDaemonSet.Spec.Template.Spec.Containers[0].Image).To(Equal(speakerImage))
+
+			By("Specify the SpeakerDaemonNodeSelector")
+			metallb.Spec.SpeakerDaemonNodeSelector = map[string]string{"node-role.kubernetes.io/worker": "true"}
+			err = k8sClient.Update(context.TODO(), metallb)
+			Expect(err).NotTo(HaveOccurred())
+
+			speakerDaemonSet = &appsv1.DaemonSet{}
+			Eventually(func() map[string]string {
+				err := k8sClient.Get(context.TODO(), types.NamespacedName{Name: consts.MetalLBDaemonsetName, Namespace: MetalLBTestNameSpace}, speakerDaemonSet)
```

Comment:
Can we check that the number of running pods changed and matches the nodes selected?